From 251eadfce55b8e5d7a2ac2d87d17ba36147cb4ad Mon Sep 17 00:00:00 2001
From: Amaury Denoyelle <adenoyelle@haproxy.com>
Date: Thu, 24 Mar 2022 17:14:52 +0100
Subject: [PATCH] MINOR: mux-quic: activate qmux traces on stdout via macro

This commit is similar to the following one :
  commit 118b2cbf8430a9513947c27a8403ff380e1dcaf2
  MINOR: quic: activate QUIC traces at compilation

If the macro ENABLE_QUIC_STDOUT_TRACES is defined, qmux traces are
outputted automatically on stdout. This is useful for the haproxy-qns
interop docker image.
---
 src/qmux_trace.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/qmux_trace.c b/src/qmux_trace.c
index 5f3bda0e1..025f91c60 100644
--- a/src/qmux_trace.c
+++ b/src/qmux_trace.c
@@ -5,6 +5,7 @@
 #include <haproxy/connection.h>
 #include <haproxy/chunk.h>
 #include <haproxy/mux_quic.h>
+#include <haproxy/sink.h>

 /* trace source and events */
 static void qmux_trace(enum trace_level level, uint64_t mask,
@@ -102,6 +102,21 @@ static void qmux_trace(enum trace_level level, uint64_t mask,
 	}
 }
 
+/* Function to automatically activate QUIC MUX traces on stdout.
+ * Activated via the compilation flag -DENABLE_QUIC_STDOUT_TRACES.
+ * Main use for now is in the docker image for QUIC interop testing.
+ */
+static void qmux_init_stdout_traces(void)
+{
+#ifdef ENABLE_QUIC_STDOUT_TRACES
+	trace_qmux.sink = sink_find("stdout");
+	trace_qmux.level = TRACE_LEVEL_DEVELOPER;
+	trace_qmux.state = TRACE_STATE_RUNNING;
+	trace_qmux.verbosity = QMUX_VERB_MINIMAL;
+#endif
+}
+INITCALL0(STG_INIT, qmux_init_stdout_traces);
+
 
 /* register qmux traces */
 INITCALL1(STG_REGISTER, trace_register_source, TRACE_SOURCE);
-- 
2.36.1

